arangod/IResearch/IResearchViewCoordinator.h

```diff
@@ -126,13 +127,24 @@ class IResearchViewCoordinator final : public LogicalView {
 
   Result renameImpl(std::string const& oldName) final;
 
+  bool isBuilding() const final;
+
   struct ViewFactory;
 
   IResearchViewCoordinator(TRI_vocbase_t& vocbase, VPackSlice info);
 
   // transient member, not persisted
-  std::unordered_map<DataSourceId, std::pair<std::string, VPackBuilder>>
-      _collections;
+  struct Data {
+    Data(std::string cname, VPackBuilder&& builder, bool building)
+        : collectionName{std::move(cname)},
+          definition{std::move(builder)},
+          isBuilding{building} {}
+
+    std::string collectionName;
+    VPackBuilder definition;
+    bool isBuilding{false};
+  };
+  containers::FlatHashMap<DataSourceId, Data> _collections;
```

Comment:
Probably NodeHashMap would work better given the size of the value type

Reply:
Done